sklearn/model_selection/_search.py

```diff
@@ -344,6 +344,42 @@ def _check_param_grid(param_grid):
                 )
 
 
+def _check_refit(search_cv, attr):
+    if not search_cv.refit:
+        raise AttributeError(
+            f"This {type(search_cv).__name__} instance was initialized with "
+            f"`refit=False`. {attr} is available only after refitting on the best "
+            "parameters. You can refit an estimator manually using the "
+            "`best_params_` attribute"
+        )
+
+
+def _estimator_has(attr):
+    """Check if we can delegate a method to the underlying estimator.
+
+    Calling a prediction method will only be available if `refit=True`. In
+    such case, we check first the fitted best estimator. If it is not
+    fitted, we check if the unfitted estimator.
+    """
+
+    def check(self):
+        _check_refit(self, attr)
+        if hasattr(self, "best_estimator_"):
+            if not hasattr(self.best_estimator_, attr):
+                raise AttributeError(
+                    f"'{type(self.best_estimator_).__name__}' object has no attribute "
+                    f"'{attr}'"
+                )
+            return True
```

Comment:
Now that I have seen so many of these. Maybe... this is nicer?

```suggestion
            # raises AttributeError if `attr` does not exist
            getattr(self.best_estimator_, attr)
            return True
```

Reply:
At least we rely on the python error formatting. This is fine with me.

Reply:
and I even understand now your logic with

```python
# returns True or raise an AttributeError if `attr` doe no exist
return True or getattr(self.best_estimator_, attr)
```